LibertyDSNP/frequency
# Event Custom RPC

# Goal
The goal of this PR is to offer an RPC endpoint with easier options for getting events and handling scale decoding.

It reduces the scale decoding issue to one of just the event enum/struct instead of needing to worry about the substrate wrapping of the events.

Closes #994

## Testing

1. Start instant sealing
2. Create schema, msa, or other things that trigger events
3. Snag the block hash and put in into this curl command (feel free to skip the `jq` call if you want non-formatted response):
```
curl --location --request POST 'localhost:9933' --header 'Content-Type: application/json' --data-raw '{
    "id":1,
    "jsonrpc":"2.0",
    "method": "frequency_getEvents",
    "params": ["'$HASH'"]
}' | jq
```

4. You should see the events and can scale decode them using the event structure and https://www.shawntabrizi.com/substrate-js-utilities/codec/

MSA Create Event SCALE
```
{
  "Msa.MsaCreated": {
    "msa_id": "u64",
    "key": "[u8; 32]"
  }
}
```

# To Do
- [x] Test js/api-augment updates

# Checklist
- [x] Updated js/api-augment.
